snapcraft/plugins/catkin.py

```diff
@@ -919,3 +961,58 @@ def _get_highest_version_path(path):
         raise RuntimeError('nothing found in {!r}'.format(path))
 
     return paths[-1]
+
+
+class _Wstool:
+    def __init__(self, ros_package_path, wstool_path, ubuntu_sources, project):
+        self._ros_package_path = ros_package_path
+        self._ubuntu_sources = ubuntu_sources
+        self._wstool_path = wstool_path
+        self._wstool_install_path = os.path.join(wstool_path, 'install')
+        self._project = project
+
+    def setup(self):
+        os.makedirs(self._wstool_install_path, exist_ok=True)
+
+        # wstool isn't a dependency of the project, so we'll unpack it
+        # somewhere else, and use it from there.
+        logger.info('Preparing to fetch wstool...')
+        ubuntu = repo.Ubuntu(self._wstool_path, sources=self._ubuntu_sources,
+                             project_options=self._project)
+        logger.info('Fetching wstool...')
+        ubuntu.get(['python-wstool'])
+
+        logger.info('Installing wstool...')
+        ubuntu.unpack(self._wstool_install_path)
+
+        logger.info('Initializing workspace (if necessary)...')
+        try:
+            self._run(['init', self._ros_package_path, '-j{}'.format(
+                self._project.parallel_build_count)])
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf8').strip()
+            if 'already is a workspace' not in output:
+                raise RuntimeError(
+                    'Error initializing workspace:\n{}'.format(output))
+
+    def merge(self, rosinstall_file):
+        return self._run(
+            ['merge', rosinstall_file, '--confirm-all', '-t{}'.format(
+                self._ros_package_path)]).strip()
+
+    def update(self):
+        return self._run(
+            ['update', '-j{}'.format(self._project.parallel_build_count),
+             '-t{}'.format(self._ros_package_path)]).strip()
+
+    def _run(self, arguments):
+        env = os.environ.copy()
+
+        # We want to make sure we use our own rosdep (which is python)
+        env['PATH'] += ':' + os.path.join(self._wstool_install_path, 'usr',
+                                          'bin')
+        env['PYTHONPATH'] = os.path.join(self._wstool_install_path, 'usr',
```

Comment:
Removing this causes wstool to not be found, so this does indeed seem to be required. Comment added.